src/Microsoft.Health.Dicom.SqlServer/Registration/Experimental/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,174 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Services.AppAuthentication;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.SqlServer;
+using Microsoft.Health.SqlServer.Configs;
+using Microsoft.Health.SqlServer.Extensions;
+using Microsoft.Health.SqlServer.Features.Client;
+using Microsoft.Health.SqlServer.Features.Storage;
+using Polly;
+using Polly.Contrib.WaitAndRetry;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class ServiceCollectionExtensions
+    {
+        private const string SqlServerSectionName = "SqlServer";
+
+        public static ISqlServiceBuilder AddSqlServer(
+            this IServiceCollection services,
+            IConfiguration configurationRoot)
+        {
+            EnsureArg.IsNotNull(services, nameof(services));
+            EnsureArg.IsNotNull(configurationRoot, nameof(configurationRoot));
+
+            var config = new SqlServerDataStoreConfiguration();
+            configurationRoot.GetSection(SqlServerSectionName).Bind(config);
+
+            services
+                .AddSingleton(config)
+                .AddConnectionServices(config);
+
+            return new SqlServiceBuilder(services, config);
+        }
+
+        private static IServiceCollection AddConnectionServices(
```

Comment:
Same as Blob, I feel changing `AddConnectionServices` to base on `ISqlServiceBuilder` is more clean, but I'm ok with it right now.

Reply:
I think these sort of things are necessary, so it may not make sense to separate out this core component unlike the others